File: backend/Fung.COMMON/DTO/FuelTank/FuelTankInfoDTO.cs
using Fung.COMMON.DTO.LevelIndicator;
using Fung.COMMON.DTO.RemainingTransaction;

namespace Fung.COMMON.DTO.FuelTank
{
    public sealed class FuelTankInfoDTO
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public int Number { get; set; }
        public double Capacity { get; set; }
        public double CurrentAmount { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int StationId { get; set; }
        public ICollection<LevelIndicatorTransactionsDTO>? LevelTransactions { get; set; }
        public ICollection<RemainingTransactionsDTO>? RemainingTransactions { get; set; }
        public string StationName { get; set; } = null!;
    }
}

File: frontend/src/models/DTO/FuelTank/fuel-tank-infoDTO.ts
import { LevelTransaction } from "src/models/Entities/level-transaction"
import { RemainingTransaction } from "src/models/Entities/remaining-transaction"

export interface FuelTankInfoDTO
{
    id: number,
    name: string,
    number: number,
    capacity: number,
    currentAmount: number,
    stationId: number,
    updatedAt: Date,
    stationName: string,
    levelTransactions?: Array<LevelTransaction>
    remainingTransactions?: Array<RemainingTransaction>
}